core/templates/dev/head/profile/Preferences.js

```diff
@@ -35,6 +35,10 @@ oppia.controller('Preferences', ['$scope', '$http', '$rootScope', '$modal', '$ti
     _saveDataItem('user_bio', userBio);
   };
 
+  $scope.saveSubjectInterests = function(subjectInterests) {
```

Comment:
You'll need to write the checking logic here. See updateWarningText() in Signup.js -- you would look at subjectInterests and ensure that everything in it is correct before calling _saveDataItem. You may need to declare a $scope.subjectInterestsWarningMessage that this method populates.